README.md

```diff
@@ -173,6 +173,38 @@ The Geolite2-City database can be downloaded from MaxMind's site at [https://dev
 Download the GeoIP2 Binary and save it at `geo_data/GeoLite2-City.mmdb`.
 The app will start using that Geolite2 file for geolocation after restart.
 
+#### Testing on a mobile device or in a virtual machine
+
+By default, the application binds to `localhost`. To test on a network device or within a virtual machine, you can bind to `0.0.0.0` instead, using the following instructions:
+
+1. Determine your computer's network IP address. On macOS, you can find this in the "Network" system settings, shown under the "Status: Connected" label. This often takes the format of `192.168.1.x` or `10.0.0.x`.
+2. In `config/application.yml`, replace `localhost` in the `domain_name` setting with the IP address discovered in the previous step. Leave the trailing port `:3000` unchanged.
+3. Start the server using the command `SMTP_HOST=192.168.1.x rails s -b 0.0.0.0`, replacing `SMTP_HOST` value with the IP address from the first step.
```

Comment:
I think ideally we'd figure out SMTP and make email work, otherwise we'll be pretty limited in what we can do in this mode of operation.

Reply:
I think the issue I was seeing was because running Rails built-in server won't start Mailcatcher, so the mailer expecting to be running on port 1025 won't be running, hence the error.

The idea described in https://github.com/18F/identity-idp/pull/4708/files#r582027295 to allow assigning the host through `make run` will indirectly resolve this.

That being said, I assume it's still a problem if one was to use the Rails built-in server. Have you noticed this being an issue in your normal usage, @mitchellhenke ?

Maybe `make run` can assign `SMTP_HOST` environment variable, then we only configure `smtp_settings` if there's an env variable present?

https://github.com/18F/identity-idp/blob/235c96d09a46ef15c27e1e025962b00df2c4970f/config/environments/development.rb#L29

Reply:
I've always started mailcatcher separately in the foreground to be able to see it doing stuff (`mailcatcher -f`). Separately, I think we could consider adding `mailcatcher` to the Gemfile so it doesn't have to be `gem install`ed.

Changing the dev SMTP config to accommodate that makes sense.

Reply:
Tested a bit locally. We'll still need to configure it, since otherwise Rails defaults to `localhost:25`.

Something like this could work:

```rb
if ENV['SMTP_HOST']
  config.action_mailer.smtp_settings = { address: ENV['SMTP_HOST'], port: 1025 }
else
  config.action_mailer.delivery_method = :test
end
```

I'm inclined to leave this for a separate pull request, since it's not strictly needed here anymore for these instructions.

Reply:
👍🏼